.github/workflows/ci.yaml

```diff
@@ -0,0 +1,159 @@
+name: CI
+
+on: push
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version-file: .nvmrc
+          cache: npm
+
+      - name: Install dependencies
+        run: npm ci --ignore-scripts
+
+      - name: Build
+        run: npm exec -- lerna run build --concurrency=2
+
+      - name: Upload dist artifacts
+        uses: actions/upload-artifact@v4
+        with:
+          name: dist-artifact
+          path: packages/**/dist
+
+  build-demo:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version-file: .nvmrc
+          cache: npm
+
+      - name: Install dependencies
+        run: npm ci --ignore-scripts
+
+      - name: Build Demo
+        run: npm run build:demo
+
+      - name: Upload demo artifacts
+        uses: actions/upload-artifact@v4
+        with:
+          name: demo-artifact
+          path: ./demo
+
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version-file: .nvmrc
+          cache: npm
+
+      - name: Install dependencies
+        run: npm ci --ignore-scripts
```

Comment:
Is there a way to run install only once, as we do with CircleCI, and subsequently use the installed `node_modules` cache for subsequent test, deploy, and publish steps?

Reply:
> Note It is not recommended to cache node_modules, as it can break across Node versions and won't work with npm ci

https://github.com/actions/cache/blob/main/examples.md#node---npm

Reply:
But, just like we currently have with CircleCI, the `npm ci` shouldn't need to run more than once. So I don't believe this statement would apply. I'd like to see this setup start by mirroring current CI as closely as possible and then work out from there.

Reply:
I'm following best-practices as defined in the [docs](https://docs.github.com/en/actions/using-workflows/storing-workflow-data-as-artifacts#comparing-artifacts-and-dependency-caching). That approach also aligns with Github Primer React's [config](https://github.com/primer/react/blob/c093411ec6caa4d4ce1720a68a22e492d5f39458/.github/workflows/ci.yml).

> Artifacts and caching are similar because they provide the ability to store files on GitHub, but each feature offers different use cases and cannot be used interchangeably.

> - Use caching when you want to reuse files that don't change often between jobs or workflow runs, such as build dependencies from a package management system.
> - Use artifacts when you want to save files produced by a job to view after a workflow run has ended, such as built binaries or build logs.

To mirror the CCI config, we would have to upload the ~1GB workspace as an artifact and then download it for every job. That will probably take longer than executing `npm ci --no-scripts` _with_ caching.

> Storing artifacts uses storage space on GitHub 

[Source](https://docs.github.com/en/actions/using-workflows/storing-workflow-data-as-artifacts#comparing-artifacts-and-dependency-caching:~:text=Storing%20artifacts%20uses%20storage%20space%20on%20GitHub)

One CI run with an stored 1GB artifact would set up above our [free 500MB monthly allowance](https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions#included-storage-and-minutes). That could become costly.

Reply:
Ok, I'm on board with the current approach.